community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/idp/IDPSolver.scala

```diff
@@ -42,6 +42,7 @@ class IDPSolver[Solvable, Result, Context](generator: IDPSolverStep[Solvable, Re
                          registryFactory: () => IdRegistry[Solvable] = () => IdRegistry[Solvable], // maps from Set[S] to BitSet
                          tableFactory: (IdRegistry[Solvable], Seed[Solvable, Result]) => IDPTable[Result] = (registry: IdRegistry[Solvable], seed: Seed[Solvable, Result]) => IDPTable(registry, seed),
                          maxTableSize: Int, // limits computation effort by reducing result quality
+                         iterationDurationLimit: Long, // limits computation effort by reducing result quality
```

Comment:
That comment sounds convoluted to me. What about `limits computation effort (and reduces result quality)`. It's not the reduction of quality which is the active part, it's the limitation, right?